owncloud/map.jinja

```diff
@@ -1,39 +1,36 @@
-{% set owncloud = salt['grains.filter_by']({
-    'Debian': {
-        'pkg': 'owncloud',
-        'repo_file': '/etc/apt/sources.list.d/owncloud.list',
-    },
-    'Ubuntu': {
-        'pkg': 'owncloud',
-        'repo_file': '/etc/apt/sources.list.d/owncloud.list',
-    },
-}, merge=salt['grains.filter_by']({
-    'vivid':{
-        'pkg_repo': 'deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_15.04/ /',
-        'key_url': 'http://download.opensuse.org/repositories/isv:ownCloud:community/xUbuntu_15.04/Release.key',
-    },
-    'utopic': {
-        'pkg_repo': 'deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_14.10/ /',
-        'key_url': 'http://download.opensuse.org/repositories/isv:ownCloud:community/xUbuntu_14.10/Release.key',
-    },
-    'trusty': {
-        'pkg_repo':  'deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_14.04/ /',
-        'key_url': 'http://download.opensuse.org/repositories/isv:ownCloud:community/xUbuntu_14.04/Release.key',
-    },
-    'saucy': {
-        'pkg_repo': 'deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_13.10/ /',
-        'key_url': 'http://download.opensuse.org/repositories/isv:ownCloud:community/xUbuntu_13.10/Release.key',
-    },
-    'precise': {
-        'pkg_repo': 'deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_12.04/ /',
-        'key_url': 'http://download.opensuse.org/repositories/isv:ownCloud:community/xUbuntu_12.04/Release.key',
-    },
-    'jessie': {
-        'pkg_repo': 'deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/Debian_8.0/ /',
-        'key_url': 'http://download.opensuse.org/repositories/isv:ownCloud:community/Debian_8.0/Release.key',
-    },
-    'wheezy': {
-        'pkg_repo': 'deb http://download.opensuse.org/repositories/isv:/ownCloud:/community/Debian_7.0/ /',
-      'key_url': 'http://download.opensuse.org/repositories/isv:ownCloud:community/Debian_7.0/Release.key',
-    },
-}, grain='oscodename', merge=salt['pillar.get']('owncloud:lookup'))) %}
+{% import_yaml "owncloud/defaults.yaml" as default_settings %}
+{% import_yaml "owncloud/osfamilymap.yaml" as osfamilymap %}
+
+{% set owncloud = salt['grains.filter_by'](
+    {'default': default_settings}
+, merge=salt['grains.filter_by'](
+    osfamilymap,
+    grain='os_family'
+, merge=salt['grains.filter_by'](
+    {
+        'default': {
+            'client_pkg_repo': "http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/{}_{}/ /".format(grains['os'], grains['osrelease']),
```

Comment:
We should have Python 2.6 support for a tad longer, so we need indexed placeholders here.

Reply:
Thanks! I'll fix that as soon as I find some time to do so.